packages/deployments/contracts/contracts/core/connext/helpers/OZERC20.sol

```diff
@@ -381,10 +381,7 @@ contract ERC20 is IERC20, IERC20Permit, EIP712 {
    */
   function DOMAIN_SEPARATOR() external view override returns (bytes32) {
     // See {EIP712._buildDomainSeparator}
-    return
-      keccak256(
-        abi.encode(_TYPE_HASH, keccak256(abi.encode(token.name)), _HASHED_VERSION, block.chainid, address(this))
-      );
+    return _domainSeparatorV4();
```

Comment:
important to note the `_HASHED_NAME` may change due to updates to name / symbol made via [`updateDetails`](https://github.com/connext/nxtp/blob/e235d66b3709e101845a8d82e762372cc1043cf0/packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol#L251-L268)

Reply:
Oeps @LayneHaber this is can of worms!
Checking the code, this issue already exists, I've documented it here:
https://github.com/spearbit-audits/connext-nxtp/issues/235